DataFormats/simulation/src/Stack.cxx

```diff
@@ -79,6 +80,26 @@ Stack::Stack(Int_t size)
   if (vmc && strcmp(vmc->GetName(), "TGeant4") == 0) {
     mIsG4Like = true;
   }
+
+  auto& param = o2::sim::StackParam::Instance();
+  if (param.inhibitPrimary.compare("none") == 0) {
+    mInhibitPrimary = [](const TParticle& p) {
+      return false;
+    };
+  } else if (param.inhibitPrimary.compare("all") == 0) {
+    mInhibitPrimary = [](const TParticle& p) {
+      return true;
+    };
+  } else if (param.inhibitPrimary.compare("forward") == 0) {
+    mInhibitPrimary = [](const TParticle& p) {
+      if (std::fabs(p.Eta()) > 1.0) {
+        return true;
+      }
+      return false;
+    };
+  } else {
+    mInhibitPrimary = [](const TParticle& p) { return false; };
+  }
```

Comment:
Hi @preghenella : I think this is good as an initial step. In the end, this could probably completely go as a fully customizable hook into the event generation phase?

Reply:
the issue is that if we put in the event generator stage we have no way to distinguish between final-state primary particles that in principle needed to be tracked but were inhibited from particles that were not asked for tracking.

Unless we pass a flag via the Stack::PushTrack (but this is in the FairStack I think)